README.md

```diff
@@ -151,7 +149,38 @@ nativeFactTypes:
 
 Note: You only need to specify the verb for the canonical for of the fact type, any synonymous forms will automatically be remapped to the canonical form
 
-
-### Tests
+## Tests
 
 Tests can be found under the `test/` folder, to run the whole suite use `npm test`
+
+## Storing files and other large objects
+
+An application can choose between two types to save file content or another large object: `File` or `WebResource`. When using a `File`, PineJS saves the content in the database using a binary data type like `BYTEA` or `BLOB`. When using a `WebResource`, PineJS saves the binary content on an external storage service and then writes metadata, including the content public URL, to the database. Client apps use the `WebResource` `href` to get the content.
+
+
+### WebResource
+
+Type [`WebResource`](./src//types/web-resource.ts) can be used to persist files or other large content on an external object storage service like Amazon S3. By "object storage" we refer to a service that can store the content and provide a URL to access that content.
+
+In order to save a `WebResource` you send an instance of [`WebResourceInput`](./src/types/web-resource.ts#L18)
+
+```js
+{
+  filename: string;
+  data: Buffer;
+  contentType?: string;
+  contentDisposition?: string;
+  size?: number;
+  storage: string;
+}
+```
+
+A typical use case is to have a web app where users can upload a file. The app will use [multer](https://github.com/expressjs/multer) to get the file from the http request as a `Buffer` and pass it as the `data` attribute of a `WebResourceInput`.
```

Comment:
The app does need to do additional transformation like following:
```ts
const multerTransform: Express.RequestHandler = async (req, _res, next) => {
	try {
		if (req.files && req.files instanceof Array) {
            req.files.forEach((file: { fieldname: string | number; }) => {
				req.body[file.fieldname] = file;
			});
		}
		next();
	} catch (err) {
		next(err);
	}
};
```

Not sure which is the best place to mention that; specifically for our documentation purpose.

Reply:
I'll add a link from this repo to PineJS, where we can put a document explaining how to handle `multipart/form-data`

Reply:
@fisehara This is to provide the basic option for a user that wants to send more than one file; it may not be a common case but it may be useful say if someone wants to upload a `README.md` resource and a `logo.png` resource

Reply:
👍